int32 newPos
---
int8 ret
